### PR TITLE
ZENREACH-7573: Replace File Identifier with Metadata

### PR DESCRIPTION
**Pull request by BlueDragonX**
[ZENREACH-7573](https://zenreach.atlassian.net/browse/ZENREACH-7573)

After initialization Kafka writes a `meta.properties` file to one of its log directories. This file contains a `broker.id` property. This PR removes support for `BROKER_ID_FILE` in favor of reading `meta.properties`.

### CODE REVIEW

File: identifier/metadata/metadata_test.go [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+				{BrokerID: 1},
+			},
+			BrokerID: 1,
+		},
+	}
+
+	for i := range tests {
+		test := tests[i]
+		t.Run(test.Name, func(t *testing.T) {
+			tempDir, err := ioutil.TempDir("", "kafka-runner-")
```
Comment (mrbroll) [inline identifier/metadata/metadata_test.go]: You can add a `t.Parallel()` in here too.

File: identifier/metadata/metadata.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	return &Identifier{
+		logDirs: logDirs,
+	}
+}
+
+// Identify the broker. Return the broker ID or an error on failure.
+func (id *Identifier) Identify() (int, error) {
+	for _, logDir := range id.logDirs {
+		brokerID, err := id.readMeta(logDir)
+		if err == nil {
```
Comment (mrbroll) [inline identifier/metadata/metadata.go]: This `err` never gets handled. There should be a non-nil check for it, and probably change this to `if brokerID != 0`.

File: identifier/metadata/metadata_test.go (hunk 1 above)
Comment (BlueDragonX) [reply identifier/metadata/metadata_test.go]: I find there's a fine line here. For heavier test cases it makes sense, but the cost of paralleling small sub-tests negates any benefit.

File: identifier/metadata/metadata.go (hunk 2 above)
Comment (BlueDragonX) [reply identifier/metadata/metadata.go]: No, this is the way I wanted this to work. There's nothing wrong with this.

Comment (mrbroll) [reply identifier/metadata/metadata.go]: But the error returned from this won't always be correct. There are several error cases in `readMeta`, but only one of them is caused by the broker id not being set.

File: identifier/metadata/metadata_test.go (hunk 1 above)
Comment (mrbroll) [reply identifier/metadata/metadata_test.go]: good point

**Pull request merged**